### PR TITLE
Add support for quoted primary key columns in uri upon import

### PR DESCRIPTION
**Pull request by strk**
See http://hub.qgis.org/issues/15226

### CODE REVIEW

**Comment by strk** [comment]
Still needs further testing (including automated) and since we're at it I could add support for importing layers with multi-column primary keys

File: src/providers/postgres/qgspostgresprovider.cpp [hunk 1]
```diff
@@ -3523,16 +3540,24 @@ QgsVectorLayerImport::ImportError QgsPostgresProvider::createEmptyLayer(
   }
   else
   {
-    // search for the passed field
-    for ( int fldIdx = 0; fldIdx < fields.count(); ++fldIdx )
+    QStringList cols = parseUriKey(primaryKey);
+    if ( cols.size() > 1 ) {
+      QgsMessageLog::logMessage( tr( "crateEmptyLayer does not support composite keys yet" ) );
```
Comment (jef-n) [inline src/providers/postgres/qgspostgresprovider.cpp]: s/crateEmptyLayer/createEmptyLayer/

File: src/providers/postgres/qgspostgresprovider.h [hunk 2]
```diff
@@ -119,6 +119,11 @@ class QgsPostgresProvider : public QgsVectorDataProvider
     static QString endianString();
 
     /**
+     * Returns a lits of unquoted column names from an uri key
```
Comment (jef-n) [inline src/providers/postgres/qgspostgresprovider.h]: s/lits/list/

**Review by jef-n — APPROVED** [review]

**Comment by strk** [comment]
Typo fixed and the "unsupported multi-column key" message removed as with new commits we now support that !

I've tried with multi-column key, a  FidMapped (string) key and a view, plus a normal table.
It's still weird that for a normal table you get an unquoted attribute in the Uri key while for a view or fid-mapped you get a quoted attribute instead, but with the code in this branch the importer would assume quotes are never part of the attribute name.

If I understood correctly these URI keys end up in project files, so we can't just change semantic w/out breaking backward compatibility ?

**Comment by strk** [comment]
I've now also added a test, but I still don't like to see those quoted identifiers in the QgsDatasourceUri.key parameter. It's good to support them for backward compatibility, but going forward I think we should avoid the quotes in the URI.

@jef-n how about reverting daa6510970e9afbc4d41d28e0c94b4f238eb372d and deal with that bug on the reading side ?

**Comment by strk** [comment]
@jef-n I have just tested, and reverting daa6510 does not reintroduce the bug in http://hub.qgis.org/issues/13710 -- clean URI with no quotes are better, aren't them ? I'll push another commit to that extent

**Comment by strk** [comment]
Pushed as 5abdfcb8064f3b3a848358afb95723b932dd3b05 and dd85796bae24842da078ad2a3e90f451839db993

**Pull request closed without merging**